### DISCUSSION TITLE
Why do I need to set this pg function security as "Definer" for it to work with RLS policies

### DISCUSSION THREAD

**Discussion by Bochelly** [supabase/supabase#26988] [Questions] [answered]
Hi,
I'm confused because I use a function in an RLS policy which only works if the security of this function is "Definer", which obviously isn't ideal. What role is used to run the policies? From the logs I added, I understand it's `authenticator` whom I authorized to read the tables just in case. 
```sql
-- USED IN RLS INSERT POLICY ON TABLE "EXPERIMENT" --
begin
 return exists (
        select a.id as admin_id
        from public.admin as a 
        left join public.company as c on a.company_id = c.id 
        where (auth.uid() = a.id) and (((extract_domain_url(new_url)) = (extract_domain_url(c.url_domain))) or (c.id = 2)) 
    );
end;
```

After some tests, it looks like the function can't select anything on `public.admin` when executed by RLS, while it works perfectly fine when security is definer. I'm probably missing something obvious, but couldn't find anything in the doc.

**Comment by GaryAustin1** [comment, marked as answer]
You need to meet RLS on every table used unless you use security definer to bypass RLS in a function.  So if the query is done by authenticated user 1 then all RLS on all tables must allow them select access to the needed rows.   There are also cases where you have an select RLS policy and it does a select on the same table.  That will cause infinite recursion because it has to meet the policy to call the policy,.
